atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraTimestampUtils.java

```diff
@@ -0,0 +1,191 @@
+import com.palantir.atlasdb.table.description.TableMetadata;
+import com.palantir.atlasdb.table.description.ValueType;
+import com.palantir.atlasdb.transaction.api.ConflictHandler;
+import com.palantir.util.Pair;
+
+public final class CassandraTimestampUtils {
+    public static final TableMetadata TIMESTAMP_TABLE_METADATA = new TableMetadata(
+            NameMetadataDescription.create(ImmutableList.of(
+                    new NameComponentDescription("timestamp_name", ValueType.STRING))),
+            new ColumnMetadataDescription(ImmutableList.of(
+                    new NamedColumnDescription(
+                            CassandraTimestampUtils.ROW_AND_COLUMN_NAME,
+                            "current_max_ts",
+                            ColumnValueDescription.forType(ValueType.FIXED_LONG)))),
+            ConflictHandler.IGNORE_ALL);
+
+    public static final String ROW_AND_COLUMN_NAME = "ts";
+    public static final String BACKUP_COLUMN_NAME = "oldTs";
+    public static final ByteString INVALIDATED_VALUE = ByteString.copyFrom(new byte[] {0});
+
+    private static final long CASSANDRA_TIMESTAMP = -1;
+
+    private static final String ROW_AND_COLUMN_NAME_HEX_STRING = encodeCassandraHexString(ROW_AND_COLUMN_NAME);
+
+    @VisibleForTesting
+    static final String APPLIED_COLUMN = "[applied]";
+    static final String COLUMN_NAME_COLUMN = "column1";
+    static final String VALUE_COLUMN = "value";
+
+    @VisibleForTesting
+    static final byte[] SUCCESSFUL_OPERATION = {1};
+
+    private CassandraTimestampUtils() {
+        // utility class
+    }
+
+    public static ByteBuffer constructSelectFromTimestampTableQuery() {
+        return toByteBuffer(String.format(
+                "SELECT %s, %s FROM %s WHERE key=%s;",
+                COLUMN_NAME_COLUMN,
+                VALUE_COLUMN,
+                wrapInQuotes(AtlasDbConstants.TIMESTAMP_TABLE.getQualifiedName()),
+                ROW_AND_COLUMN_NAME_HEX_STRING));
+    }
+
+    public static ByteBuffer constructCheckAndSetMultipleQuery(Map<String, Pair<byte[], byte[]>> checkAndSetRequest) {
+        StringBuilder builder = new StringBuilder();
+        builder.append("BEGIN UNLOGGED BATCH\n"); // Safe, because all updates are on the same partition key
+
+        // Safe, because ordering does not apply in batches
+        checkAndSetRequest.entrySet().forEach(entry -> {
+            String columnName = entry.getKey();
+            byte[] expected = entry.getValue().getLhSide();
+            byte[] target = entry.getValue().getRhSide();
+            builder.append(constructCheckAndSetQuery(columnName, expected, target));
+        });
+        builder.append("APPLY BATCH;");
+        return toByteBuffer(builder.toString());
+    }
+
+    public static boolean wasOperationApplied(CqlResult cqlResult) {
+        byte[] applied = getNamedColumnValue(getColumnsFromOnlyRow(cqlResult), APPLIED_COLUMN);
+        return Arrays.equals(applied, SUCCESSFUL_OPERATION);
```

Comment:
Copypasting the result I used when explaining it to Glenn above: your result looks something like this

```java
 [applied] | key    | column1 | column2 | value
-----------+--------+---------+---------+--------------------
     False | 0x7473 |  0x7473 |      -1 | 0x00000000000f6951
```